llvm/lib/CodeGen/TwoAddressInstructionPass.cpp

```diff
@@ -1937,13 +1937,16 @@ eliminateRegSequence(MachineBasicBlock::iterator &MBBI) {
   }
 
   bool DefEmitted = false;
+  bool DefIsPartial = false;
   for (unsigned i = 1, e = MI.getNumOperands(); i < e; i += 2) {
     MachineOperand &UseMO = MI.getOperand(i);
     Register SrcReg = UseMO.getReg();
     unsigned SubIdx = MI.getOperand(i+1).getImm();
     // Nothing needs to be inserted for undef operands.
-    if (UseMO.isUndef())
+    if (UseMO.isUndef()) {
+      DefIsPartial = true;
```

Comment:
Do we actually know a REG_SEQUENCE fully covers the result register? e.g. something like 

`%2:vreg_128 = REG_SEQUENCE %0, %subreg.sub0, %1, %subreg.sub1`

is valid but doesn't define all lanes

Reply:
Yes, those should have already been handled.